src/wp-includes/user.php

```diff
@@ -2221,8 +2221,20 @@ function wp_insert_user( $userdata ) {
 		$user_pass = ! empty( $userdata['user_pass'] ) ? $userdata['user_pass'] : $old_user_data->user_pass;
 	} else {
 		$update = false;
-		// Hash the password.
-		$user_pass = wp_hash_password( $userdata['user_pass'] );
+
+		if ( empty( $userdata['user_pass'] ) ) {
+			wp_trigger_error(
+				__FUNCTION__,
+				__( 'The user_pass field is required when creating a new user. A random password has been generated.' ),
+				E_USER_WARNING
+			);
+
+			// Generate and hash a random password
+			$user_pass = wp_hash_password( wp_generate_password( 32 ) );
+		} else {
+			// Hash the password.
+			$user_pass = wp_hash_password( $userdata['user_pass'] );
+		}
```

Comment:
Although I suggested the random password on the ticket, I was wrong.

WordPress prevents authentication if the password is empty so by setting the value to an empty string, the password reset flow is forced for the user. I think this is a better option than a random password.

In `wp_authenticate_username_password()`
https://github.com/WordPress/wordpress-develop/blob/c2de959ca869d422776941a5226b672389230029/src/wp-includes/user.php#L174-L176

In `wp_authenticate_email_password()`
https://github.com/WordPress/wordpress-develop/blob/c2de959ca869d422776941a5226b672389230029/src/wp-includes/user.php#L264-L266


```suggestion
		if ( empty( $userdata['user_pass'] ) ) {
			wp_trigger_error(
				__FUNCTION__,
				__( 'The user_pass field is required when creating a new user. The user will need to reset their password before logging in.' ),
				E_USER_WARNING
			);

			// Set the password as an empty string to force the password reset flow.
			$userdata['user_pass'] = '';
		}

		// Hash the password.
		$user_pass = wp_hash_password( $userdata['user_pass'] );
```